linker/src/main/scala/za/co/absa/spline/linker/boundary/LegacyInletV5Controller.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.linker.boundary
+
+import com.mongodb.DuplicateKeyException
+import io.swagger.annotations.ApiOperation
+import org.slf4s.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.bind.annotation._
+import salat.grater
+import za.co.absa.spline.linker.control.{BatchDataLineageLinker, LineageProjectionMerger}
+import za.co.absa.spline.model.DataLineage
+import za.co.absa.spline.model.streaming.ProgressEvent
+import za.co.absa.spline.persistence.api.{DataLineageReader, DataLineageWriter, ProgressEventWriter}
+import za.co.absa.spline.persistence.mongo.serialization.BSONSalatContext._
+
+import scala.concurrent.ExecutionContext.Implicits._
+import scala.concurrent.Future
+
+@RestController
+@RequestMapping(Array("/legacyInlet/v5/"))
+@Autowired()
+class LegacyInletV5Controller(
+    lineageWriter: DataLineageWriter,
+    lineageReader: DataLineageReader,
+    progressEventWriter: ProgressEventWriter) {
+
+  private val log = LoggerFactory.getLogger(getClass)
+  private val batchLinker = new BatchDataLineageLinker(lineageReader)
+
+  @PostMapping(Array("dataLineage"))
+  @ApiOperation("Links and stores a DataLineage v5 in BSON serialized via Salat.")
+  def lineage(@RequestBody body: Array[Byte]): Future[Void] = {
+    val rawLineage = grater[DataLineage].fromBSON(body)
+    val processed = LineageProjectionMerger(rawLineage)
+    for {
+      linked <- batchLinker(processed)
+       _ <- lineageWriter.store(linked)
+         .recover({ case e: DuplicateKeyException =>
+           log.warn("Ignored DuplicateKeyException to tolerate duplicate lineage insert to MongoDB.", e)
+         }).andThen({ case _ => log.info(s"Saved lineage ${processed.id}")})
+    } yield null
```

Comment:
@wajda There may be a way to avoid using ```null``` and ```Void``` here and replace it with Scala's ```Unit```. I noticed your ```Unit``` adapter below, but when I tried to use it, the client kept interpreting the response as failure. But it is probably not worth spending much time on as current solution is quite readable and it would be just an aesthetic change.